src/magic_enchantment.cpp

```diff
@@ -462,32 +490,93 @@ void enchantment::force_add( const enchantment &rhs )
     }
 }
 
-void enchantment::set_has( enchantment::has value )
+void enchant_cache::force_add( const enchantment &rhs, const Character &guy )
+{
+    dialogue d( get_talker_for( guy ), nullptr );
+    for( const std::pair<const enchant_vals::mod, int_or_var<dialogue>> &pair_values :
+         rhs.values_add ) {
+        values_add[pair_values.first] += pair_values.second.evaluate( d );
+    }
+    for( const std::pair<const enchant_vals::mod, int_or_var<dialogue>> &pair_values :
+         rhs.values_multiply ) {
+        // values do not multiply against each other, they add.
+        // so +10% and -10% will add to 0%
+        values_multiply[pair_values.first] += 0.01 * static_cast<double>( pair_values.second.evaluate(
```

Comment:
The cast seems unnecessary. It will be done by the compiler anyway in order to multiply with the double literal `0.1`.